runtime/syntax/cabal.vim

```diff
@@ -23,83 +24,93 @@ if exists("b:current_syntax")
   finish
 endif
 
-syn keyword	cabalCategory	Library library Executable executable Flag flag
-syn keyword	cabalCategory	source-repository Source-Repository
+syn match	cabalCategory	"\c\<executable\>"
+syn match	cabalCategory	"\c\<library\>"
+syn match	cabalCategory	"\c\<benchmark\>"
+syn match	cabalCategory	"\c\<test-suite\>"
+syn match	cabalCategory	"\c\<source-repository\>"
+syn match	cabalCategory	"\c\<flag\>"
+syn match	cabalCategory	"\c\<custom-setup\>"
 
 syn keyword     cabalConditional    if else
 syn match       cabalOperator       "&&\|||\|!\|==\|>=\|<="
 syn keyword     cabalFunction       os arche impl flag
 syn match       cabalComment    /--.*$/
-syn match       cabalVersion    "\d\+\(.\(\d\)\+\)\+"
+syn match       cabalVersion    "\d\+\(\.\(\d\)\+\)\+\(\.\*\)\?"
 
-syn match       cabalTruth      "\ctrue"
-syn match       cabalTruth      "\cfalse"
+syn match       cabalTruth      "\c\<true\>"
```

Comment:
Instead of `syn match`, it's better to use `syn keyword`, the same applies to all the other changes which match a single word.i